### ISSUE TITLE
Ability to easily autolink to PRs in changelogs

### ISSUE THREAD

**Issue by wyardley** [python-semantic-release/python-semantic-release#953]
## Description
It would be nice to have links to PRs (`#123`), such as when PRs are squash merged, be able to link to the GH issue vs. just linking to the commit. Way back, I tweaked this in #292, but seems like all of that has since been updated / revamped completely.

(it would also be nice to have a regex to be able to auto-populate links to ticketing systems via some kind of regex, kind of like GH autolinks, but less important)

I can see `pull_request_url` etc. here:
https://python-semantic-release.readthedocs.io/en/latest/changelog_templates.html
However, it would be nice to at least have some docs that include more examples of custom templates; also, I'm guessing the `(#NNN)` would also show up in the message itself unless it were edited out.

## Use cases
This is also already supported out of the box in the original semantic-release implementation, and used to work in this project as well.

While I know there is templating, and would be happy to see a trivial solution to implement this natively with templating, my suggestion is that the resulting solution might be hard to read.

**Comment by codejedi365** [comment]
@wyardley, thank you for your request. I have been focused on improving changelogs as I have joined this project and have targeted this issue as well.  Unfortunately, other higher priority items have prevented me from finishing my changelog refactors but I recommend you checkout PR #866 for what I have accomplished so far.  It does include a MR/PR url replacement macro to do the item above.  It might not be the end solution as people with other custom templates will loose out on the functionality unless they do a raw copy of the macros file.
